### PR TITLE
chore(sample): Update loop_parameter.py sample so it's compatible in both v1 and v2

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by chensun** [comment]
@Bobgy the v2 sample test times out often. should we increase wait time?
https://github.com/kubeflow/pipelines/pull/5562

File: samples/core/loop_parameter/loop_parameter_test.py [hunk 1]
```diff
@@ -17,29 +17,8 @@
...
-    #   "lastUpdateTimeSinceEpoch":"1617689193656"
-    # }
-    # TestCase(
-    #     pipeline_func=my_pipeline,
-    #     mode=kfp.dsl.PipelineExecutionMode.V2_COMPATIBLE,
-    #     verify_func=NEEDS_A_FIX,
-    # ),
     TestCase(
         pipeline_func=my_pipeline,
-        mode=kfp.dsl.PipelineExecutionMode.V1_LEGACY,
```
Comment (Bobgy) [inline samples/core/loop_parameter/loop_parameter_test.py:43]: @chensun I think we want to keep testing the sample works for both V1 and V2, so there should be two test cases here.

Comment (chensun) [reply samples/core/loop_parameter/loop_parameter_test.py:43]: I see. I misunderstood. Fixed.

**Pull request merged**